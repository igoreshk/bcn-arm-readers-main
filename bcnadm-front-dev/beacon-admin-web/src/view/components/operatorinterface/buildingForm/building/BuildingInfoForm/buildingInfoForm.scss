.buildingInfoForm {
  .firstBlock {
    display: flex;
    align-items: center;
    justify-content: space-between;

    @media screen and (max-width: 650px) {
      flex-direction: column;
      align-items: unset;
    }

    .dropzone {
      width: 250px;
      height: 150px;
      border: dashed 1px;
      border-color: #999999;
      cursor: pointer;
      font-size: 13px;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      text-align: center;
      margin-left: 20px;

      @media screen and (max-width: 650px) {
        width: 100%;
        margin-left: 0;
        margin-top: 10px;
      }
    }

    .preview {
      width: inherit;
      height: inherit;
      object-fit: cover;
    }

    .error {
      color: #FF0000;
      margin-top: 20px;
    }
  }

  .secondBlock {
    .row {
      display: flex;
      white-space: nowrap;

      @media screen and (max-width: 600px) {
        flex-direction: column;
      }
            
      > div:first-child {
        margin-right: 20px;

        @media screen and (max-width: 600px) {
          margin-right: 0;
        }
      }
    }
  }

  .thirdBlock {
    .row {
      display: flex;
      white-space: nowrap;

      @media screen and (max-width: 600px) {
        flex-direction: column;
      }
    }

    .map {
      position: relative;
      width: 100%;
      height: 300px;
      margin-right: 0;
    }
  }

  .buttonsContainer {
    display: flex;
    justify-content: flex-end;
    margin-top: 20px;

    .nextButton {
      margin-left: 10px;
    }
  }
}