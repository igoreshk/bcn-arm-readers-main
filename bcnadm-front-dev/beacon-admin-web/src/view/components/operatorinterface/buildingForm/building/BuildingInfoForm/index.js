import ImageDropzone from 'src/view/components/common/ImageDropzone';
import { maxFileSize } from 'src/view/FileConsts';
import React from 'react';
import { Field } from 'redux-form';
import { Button } from '@material-ui/core';
import { FaBuilding as BuildingIcon } from 'react-icons/fa';
import PropTypes from 'prop-types';

import BuildingInfoTextField from '../BuildingInfoTextField';
import Logo from '../../styles/Logo';
import BuildingPickerMap from '../BuildingPickerMap/BuildingPickerMap';

import './buildingInfoForm.scss';

const buildingInfoForm = ({ next, allowedToProceed, imageLink, translate, isNew, closeDialog }) => (
  <div className="buildingInfoForm">
    <div className="firstBlock">
      <div>
        <Logo
          title={translate(isNew ? 'building.create.title' : 'building.update.title')}
          icon={<BuildingIcon className="logo" />}
        />
        <Field name="name" component={BuildingInfoTextField} label={translate('building.name')} maxLength={70} />
      </div>
      <Field
        component={ImageDropzone}
        name="mapImage"
        imageLink={imageLink}
        maxSize={maxFileSize}
        translate={translate}
      />
    </div>

    <div className="secondBlock">
      <Field name="address" component={BuildingInfoTextField} label={translate('building.address')} maxLength={256} />
      <div className="row">
        <Field name="phoneNumber" component={BuildingInfoTextField} label={translate('building.phoneNumber')} />
        <Field name="workingHours" component={BuildingInfoTextField} label={translate('building.workingHours')} />
      </div>
      <div className="row">
        <Field name="latitude" type="number" component={BuildingInfoTextField} label={translate('building.lat')} />
        <Field name="longitude" type="number" component={BuildingInfoTextField} label={translate('building.lng')} />
      </div>
      <div className="row">
        <Field name="height" type="number" component={BuildingInfoTextField} label={translate('building.height')} />
        <Field name="width" type="number" component={BuildingInfoTextField} label={translate('building.width')} />
      </div>
    </div>
    <div className="thirdBlock">
      <div className="row">
        <div className="map">
          <BuildingPickerMap />
        </div>
      </div>
    </div>

    <div className="buttonsContainer">
      <Button onClick={closeDialog}>{translate('building.cancel')}</Button>
      <Button onClick={next} disabled={!allowedToProceed} variant="contained" color="primary" className="nextButton">
        {translate('building.next')}
      </Button>
    </div>
  </div>
);

buildingInfoForm.propTypes = {
  translate: PropTypes.func.isRequired,
  next: PropTypes.func.isRequired,
  allowedToProceed: PropTypes.bool.isRequired,
  closeDialog: PropTypes.func.isRequired,
  imageLink: PropTypes.string,
  isNew: PropTypes.bool
};

export default buildingInfoForm;
