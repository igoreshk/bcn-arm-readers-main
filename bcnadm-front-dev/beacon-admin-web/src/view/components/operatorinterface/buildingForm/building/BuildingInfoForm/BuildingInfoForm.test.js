import React from 'react';
import { shallow } from 'enzyme';

import BuildingInfoForm from './index';

const props = {
  translate: jest.fn().mockReturnValue(''),
  next: jest.fn(),
  allowedToProceed: true,
  goTo: jest.fn(),
  setBuildingToEdit: jest.fn(),
  closeDialogForm: jest.fn(),
  isBuildingInitialized: true,
  toggleEdited: jest.fn(),
  imageLink: 'imageLink',
  type: 'number',
  closeDialog: jest.fn()
};

const wrapper = shallow(<BuildingInfoForm {...props} />);

describe('BuildingInfoForm test', () => {
  it('check props matches', () => expect(wrapper.length).toEqual(1));
  it('snapshot for BuildingInfoForm', () => expect(wrapper).toMatchSnapshot());
});
