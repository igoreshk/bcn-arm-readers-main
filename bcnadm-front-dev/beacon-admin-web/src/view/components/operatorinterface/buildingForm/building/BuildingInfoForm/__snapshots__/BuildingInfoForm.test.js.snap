// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingInfoForm test snapshot for BuildingInfoForm 1`] = `
<div
  className="buildingInfoForm"
>
  <div
    className="firstBlock"
  >
    <div>
      <Logo
        icon={
          <FaBuilding
            className="logo"
          />
        }
        title=""
      />
      <Field
        component={[Function]}
        label=""
        maxLength={70}
        name="name"
      />
    </div>
    <Field
      component={[Function]}
      imageLink="imageLink"
      maxSize={1048576}
      name="mapImage"
      translate={
        [MockFunction] {
          "calls": Array [
            Array [
              "building.update.title",
            ],
            Array [
              "building.name",
            ],
            Array [
              "building.address",
            ],
            Array [
              "building.phoneNumber",
            ],
            Array [
              "building.workingHours",
            ],
            Array [
              "building.lat",
            ],
            Array [
              "building.lng",
            ],
            Array [
              "building.height",
            ],
            Array [
              "building.width",
            ],
            Array [
              "building.cancel",
            ],
            Array [
              "building.next",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
            Object {
              "type": "return",
              "value": "",
            },
          ],
        }
      }
    />
  </div>
  <div
    className="secondBlock"
  >
    <Field
      component={[Function]}
      label=""
      maxLength={256}
      name="address"
    />
    <div
      className="row"
    >
      <Field
        component={[Function]}
        label=""
        name="phoneNumber"
      />
      <Field
        component={[Function]}
        label=""
        name="workingHours"
      />
    </div>
    <div
      className="row"
    >
      <Field
        component={[Function]}
        label=""
        name="latitude"
        type="number"
      />
      <Field
        component={[Function]}
        label=""
        name="longitude"
        type="number"
      />
    </div>
    <div
      className="row"
    >
      <Field
        component={[Function]}
        label=""
        name="height"
        type="number"
      />
      <Field
        component={[Function]}
        label=""
        name="width"
        type="number"
      />
    </div>
  </div>
  <div
    className="thirdBlock"
  >
    <div
      className="row"
    >
      <div
        className="map"
      >
        <Connect(Wrapper) />
      </div>
    </div>
  </div>
  <div
    className="buttonsContainer"
  >
    <WithStyles(ForwardRef(Button))
      onClick={[MockFunction]}
    />
    <WithStyles(ForwardRef(Button))
      className="nextButton"
      color="primary"
      disabled={false}
      onClick={[MockFunction]}
      variant="contained"
    />
  </div>
</div>
`;
